### PR TITLE
Automatially add IC Account to Account Assignments Access Requests

### PR DESCRIPTION
**Pull request by tcsc**
The UI needs access to the account associated with an Account Assignment
in order to display the enclosing Account, otherwise the user will not
be able to see their assogned permission sets.

This patch automatically adds the enclosing account for any account
assignments in a resource access request and allows the user to see
ther human-friendly names in the access request listing.

### CODE REVIEW

File: lib/auth/auth.go [hunk 1]
```diff
@@ -5282,6 +5246,66 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
...
+		if icAccounts.Contains(asmt.Spec.AccountId) {
+			continue
+		}
+
+		resources = append(resources, types.ResourceID{
+			ClusterName: resource.ClusterName,
+			Kind:        types.KindIdentityCenterAccount,
+			Name:        asmt.Spec.AccountId,
+		})
+		icAccounts.Add(asmt.Spec.AccountId)
```
Comment (smallinsky) [inline lib/auth/auth.go:5303]:
```suggestion
		icAccounts.Add(asmt.GetSpec().GetAccountId())
```

File: lib/auth/auth.go [hunk 2]
```diff
@@ -5282,6 +5246,66 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
...
+		}
+
+		if icAccounts.Contains(asmt.Spec.AccountId) {
+			continue
+		}
+
+		resources = append(resources, types.ResourceID{
+			ClusterName: resource.ClusterName,
+			Kind:        types.KindIdentityCenterAccount,
+			Name:        asmt.Spec.AccountId,
```
Comment (smallinsky) [inline lib/auth/auth.go:5301]:
```suggestion
			Name:        asmt.GetSpec().GetAccountId(),
		}),
```

File: lib/auth/auth.go [hunk 3]
```diff
@@ -5282,6 +5246,66 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
...
+
+	icAccounts := utils.NewSet[string]()
+	for _, resource := range accountAssignments {
+		assignmentID := services.IdentityCenterAccountAssignmentID(resource.Name)
+		asmt, err := a.Services.IdentityCenter.GetAccountAssignment(ctx, assignmentID)
+		if err != nil {
+			return nil, trace.Wrap(err, "fetching identity center account assignment")
+		}
+
+		if icAccounts.Contains(asmt.Spec.AccountId) {
```
Comment (smallinsky) [inline lib/auth/auth.go:5294]:
```suggestion
		if icAccounts.Contains(asmt.GetSpec().GetAccountId()) {
```

File: lib/auth/auth.go [hunk 4]
```diff
@@ -5282,6 +5246,66 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
...
+			continue
+		}
+
+		resources = append(resources, types.ResourceID{
+			ClusterName: resource.ClusterName,
+			Kind:        types.KindIdentityCenterAccount,
```
Comment (smallinsky) [inline lib/auth/auth.go:5300]: Could you leave a short comment explaining why `KindIdentityCenterAccountAssignment` requires an additional `KindIdentityCenterAccount` resource ?

**Review by smallinsky — APPROVED** [review]

File: lib/auth/auth.go [hunk 5]
```diff
@@ -5282,6 +5246,66 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
 	return req, nil
 }
 
+// appendAdditionalResources examines the set of requested resources and adds
+// any extra resources that are implicitly required by the request.
+func (a *Server) appendAdditionalResources(ctx context.Context, resources []types.ResourceID) ([]types.ResourceID, error) {
```
Comment (flyinghermit) [inline lib/auth/auth.go:5251]:
```suggestion
// appendImplicitlyRequiredResources examines the set of requested resources and adds
// any extra resources that are implicitly required by the request.
func (a *Server) appendImplicitlyRequiredResources(ctx context.Context, resources []types.ResourceID) ([]types.ResourceID, error) {
```

File: lib/auth/auth.go [hunk 6]
```diff
@@ -5130,47 +5130,11 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
...
-			}
-		}
-	}
-
-	if len(additionalResources) > 0 {
-		requestedResourceIDs = append(requestedResourceIDs, additionalResources...)
-		req.SetRequestedResourceIDs(requestedResourceIDs)
+	requestedResourceIDs, err := a.appendAdditionalResources(ctx, req.GetRequestedResourceIDs())
+	if err != nil {
+		return nil, trace.Wrap(err, "adding additional required resources")
```
Comment (flyinghermit) [inline lib/auth/auth.go:5135]:
```suggestion
		return nil, trace.Wrap(err, "adding additional implicitly required resources")
```

**Review by flyinghermit — APPROVED** [review]

**Pull request merged**